### PR TITLE
Homework submission

### PR DESCRIPTION
**Pull request by dumplingpuff**

### CODE REVIEW

**Comment by dumplingpuff** [comment]
Incomplete homework.

**Comment by dumplingpuff** [comment]
Expressworks homework.

File: expressworks/07-json-me.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    try {
+      thing = JSON.parse(data);
+    }
+    catch(error) {
+      res.sendStatus(500);
+    }
+    res.json(thing);
+  });
+});
+app.listen(process.argv[2]);
```
Comment (GA-MEB) [inline expressworks/07-json-me.js:22]: Very nice! You set this up well.

File: learnyounode/01-baby-steps.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+'use strict';
+// let result  = 0;
+//
+//   for(let i = 2; i < process.argv.length; i++) {
+//     result+= Number(process.argv[i]);
+// };
+//
+// console.log(result);
+
+console.log(process.argv);
```
Comment (GA-MEB) [inline learnyounode/01-baby-steps.js:10]: I think you wanted the sum. Here's one way to get it.

```
process.argv.slice(2).map(Number).reduce((a,b) => a + b);
```

File: learnyounode/03-my-first-async-io.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    num = arr.length - 1;
+    callback();
+  });
+}
+
+function logNum() {
+  console.log(num);
+}
+
+read(logNum);
```
Comment (GA-MEB) [inline learnyounode/03-my-first-async-io.js:18]: Nice separation of concerns - didn't expect that.

File: learnyounode/06-http-client.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+let http = require('http');
+
+http.get(process.argv[2], function(res) {
+  res.setEncoding('utf8');
+  res.on('data', function(chunk) {
+    console.log(chunk);
+  });
```
Comment (GA-MEB) [inline learnyounode/06-http-client.js:9]: Mmm.. That's not quite right. You'll want to collect all of the chunks of data and concatenate them before printing them out.

**Comment by GA-MEB** [comment]
Good job on this.

**Pull request closed without merging**